ui_framework/doc_site/src/views/expression/expression.js

```diff
@@ -1,26 +1,139 @@
-/* eslint-disable */
+import React from 'react';
+import _ from 'lodash';
 
-let isPopoverCollapsed = true;
-const $toggleExpressionButton = $('[data-id="toggleExpressionButton"]');
-const $toggleExpressionPopover = $('[data-id="toggleExpressionPopover"]');
+import {
+  KuiExpressionItem,
+  KuiExpressionItemButton,
+  KuiExpressionItemPopover,
+} from '../../../../components';
 
-$('[data-id="toggleExpressionButton"]').click(function() {
 
-  if (isPopoverCollapsed) {
-    $(this).addClass('kuiExpressionItem__button--isActive')
-    $(this).parent( "div" ).find('[data-id="toggleExpressionPopover"]').removeClass('kuiExpressionItem__popover--isHidden');
-  } else {
-    $(this).removeClass('kuiExpressionItem__button--isActive')
-    $(this).parent( "div" ).find('[data-id="toggleExpressionPopover"]').addClass('kuiExpression__popover--isHidden');
+class KuiExpressionItemExample extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      example1: {
+        value: 'count()'
+      },
+      example2: {
+        object: 'A',
+        value: '100',
+        description: 'Is above'
+      },
+      activeButton: null
+    };
+
+    this.changeExample1 = this.changeExample1.bind(this);
+    this.changeExample2Object = this.changeExample2Object.bind(this);
+    this.changeExample2Description = this.changeExample2Description.bind(this);
+    this.changeExample2Value = this.changeExample2Value.bind(this);
+    this.onOutsideClick = this.onOutsideClick.bind(this);
```

Comment:
The following style avoids having to manually bind in this way:

```
changeExample1: event => {
  this.setState(_.merge(this.state,{ example1:{ value: event.target.value } }));
}
```

Reply:
@zinckiwi Class properties are cool!